parl/remote/heartbeat_server.py

```diff
@@ -0,0 +1,73 @@
+#   Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import threading
+import zmq
+from parl.remote import remote_constants
+from parl.remote.zmq_utils import create_server_socket, create_client_socket
+from parl.utils import get_ip_address, to_byte, logger
+
+
+class HeartbeatServerThread(threading.Thread):
+    def __init__(self, heartbeat_client_recv_addr, server_type, client_type):
+        """Create a thread to run the heartbeat server.
```

Comment:
add the comment:
Note that the process will exit directly if the heartbeat detection fails.

Reply:
done